### PR TITLE
feat: added single product page

### PR DESCRIPTION
**Pull request by mariyasada**
-added a single-product page
- display all the details of the product

https://user-images.githubusercontent.com/77275100/162697080-49fbf8e1-676e-434c-80a8-9282558dce27.mp4

### CODE REVIEW

File: src/components/HorizontalCard/HorizontalCard.jsx [hunk 1]
```diff
@@ -35,7 +36,9 @@ export const HorizontalCard = ({ item }) => {
   return (
     <div className="horizontal-card pd-1 border-round">
       <div className="card-image-holder">
-        <img src={image} className="card-image" alt="product-Image" />
+        <Link to={`/product/${item._id}`}>
```
Comment (ankit9015) [inline src/components/HorizontalCard/HorizontalCard.jsx:39]: I like that you are using dynamic routes

File: src/components/singleProductCard/SingleProductCard.jsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    fastDelivery,
+    brand,
+    quantity,
+    description,
+    isSoldOut,
+  } = Product;
+  console.log(image);
+
+  const showToast = () => {
+    toast("inform you when product back in stock", { icon: "✔️" });
```
Comment (ankit9015) [inline src/components/singleProductCard/SingleProductCard.jsx:33]: I like that you are using toast to notify about availability of products in stock

File: src/components/singleProductCard/SingleProductCard.jsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            <BiRupee />
+            {originalPrice}
+          </p>
+          <p className="discount-tag-for-single-product text-size-md">
+            ({discountOffer})
+          </p>
+          <span className="rating-badge rating-badge-single-product">
+            <div
+              className={
+                isRatingGood
```
Comment (ankit9015) [inline src/components/singleProductCard/SingleProductCard.jsx:63]: I like that you are using different rating styles on a basis of good rating or not

File: src/pages/SingleProductpage/SingleProduct.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import "./SingleProduct.css";
+import { SingleProductCard } from "../../components/singleProductCard/SingleProductCard";
+import { useParams } from "react-router-dom";
+import { useProduct } from "../../context/product-context";
+
+export const SingleProduct = () => {
+  const { productList } = useProduct();
+  const { productId } = useParams();
+  console.log(productId, "useparam");
```
Comment (ankit9015) [inline src/pages/SingleProductpage/SingleProduct.jsx]: remove console.log

**Review by ankit9015 — APPROVED** [review]
PR is good to merge, just remove console.log as suggested,

File: src/components/singleProductCard/SingleProductCard.jsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+import { useCartandWishList } from "../../context/CartAndWishlist-context";
+import toast from "react-hot-toast";
+
+export const SingleProductCard = ({ Product }) => {
+  const { cartState, cartDispatch, wishListState, wishListDispatch } =
+    useCartandWishList();
+
+  const { cartItem } = cartState;
+  const { wishListItem } = wishListState;
+  console.log(Product, "singlecard");
```
Comment (t007rushi) [inline src/components/singleProductCard/SingleProductCard.jsx:15]: console here

**Review by t007rushi — APPROVED** [review]
well written code for dynamic routing

**Pull request merged**